lib/galaxy_test/selenium/test_workflow_rerun.py

```diff
@@ -0,0 +1,89 @@
+from galaxy_test.base.workflow_fixtures import WORKFLOW_SIMPLE_CAT_TWICE
+from .framework import (
+    managed_history,
+    retry_assertion_during_transitions,
+    RunsWorkflows,
+    selenium_test,
+    SeleniumTestCase,
+    UsesHistoryItemAssertions,
+)
+
+
+class TestWorkflowRun(SeleniumTestCase, UsesHistoryItemAssertions, RunsWorkflows):
+    ensure_registered = True
+
+    @selenium_test
+    @managed_history
+    def test_workflow_rerun(self):
+        # Run a workflow first
+        self.perform_upload(self.get_filename("1.fasta"))
+        self.wait_for_history()
+        self.workflow_run_open_workflow(WORKFLOW_SIMPLE_CAT_TWICE)
+        self.screenshot("workflow_run_before_rerun_ready")
+        self.workflow_run_submit()
+        self.sleep_for(self.wait_types.UX_TRANSITION)
+        self.workflow_run_wait_for_ok(hid=2, expand=True)
+        self.assert_item_summary_includes(2, "2 sequences")
+
+        # Keep track of the original history name
+        original_history_name = self.history_panel_name()
+
+        # Create a new history, different from the original
+        self.history_panel_create_new()
+        self.home()
+        self._assert_history_name_is("Unnamed history")
+        assert self.history_panel_name() != original_history_name
+
+        self.screenshot("workflow_run_before_rerun_complete")
+
+        # Route to rerun
+        self._invocations_panel_open_latest_invocation()
+        self.components.invocations.workflow_rerun_button.wait_for_and_click()
+        self.sleep_for(self.wait_types.UX_TRANSITION)
+
+        # Arriving on the rerun page, the history should have switched back to the original history
```

Comment:
Oh, we should not switch the history. What's the reason for that ?

Reply:
We switch to the history because the workflow run form loads input datasets and collections from the current history.

Therefore, if the request state contains values from another history, we need to switch that history, so that the FormData components have the original inputs.

Reply:
We switch to the history because the workflow run form loads input datasets and collections from the current history.

Therefore, if the request state contains values from another history, we need to switch that history, so that the FormData components have the original inputs.

Reply:
What is the reason for that ? We should not cement a bug with tests.

Reply:
Jobs can be rerun from a different history. That uses the same form data component, why does that not work here ?

Reply:
Thats true. 

Let me rework how the workflow run form handles this then...

Reply:
Well, that might sprawl. How about we issue a confirmation dialog if the history has to be switched for now ?

Reply:
Sure, would it be fine to instead of opening a modal, do it right where the form would appear in the center?

Because to make this work, I made it load the history FIRST, then render the form.

Reply:
> Eventually I think we should be able to switch the current history without affecting the values selected in the run form.

I think that should definitely be the thing in the future.

It's not obvious to the average user that changing the history will make them lose all of their formdata values

Reply:
> I made it load the history FIRST, then render the form.

What I was thinking is that we pop open the confirmation dialog as you click on rerun ? Since at that point you already know if you need to switch the history or not to do a rerun

Reply:
Oh that can work too yes, on cancel you stay at the invocation view, on confirm, you route to rerun (where we change history).

Reply:
Does this work?

There is one thing though. In this case, if the rerun history is different, and the user does not want to change their current history, all they can do is just cancel. What if they want to run the same workflow here but for a different history? 
To allow that I could use a custom modal instead of the confirm dialog (which only has confirm/cancel) with a 3rd option of just run workflow (instead of rerun with original inputs).

https://github.com/user-attachments/assets/af39b71a-9a72-4c92-807c-919eedc6a932

Reply:
Yeah, three options or split the button into "Run workflow" and "Rerun workflow" ? That seems fine.

Reply:
![image](https://github.com/user-attachments/assets/b6e26138-e60f-44cc-9396-41df4a338708)

Added this, will push once I adjust the selenium